site/utils/apiUtils/createDisruptionValidations.ts

```diff
@@ -0,0 +1,278 @@
+import {
+    EnvironmentReason,
+    EquipmentReason,
+    MiscellaneousReason,
+    PersonnelReason,
+} from "@create-disruptions-data/shared-ts/siriTypes";
+import dayjs from "dayjs";
+import customParseFormat from "dayjs/plugin/customParseFormat";
+import { NextApiResponse } from "next";
+import { DISRUPTION_TYPES, CD_DATE_FORMAT } from "../../constants/index";
+import { ErrorInfo } from "../../interfaces";
+import { redirectTo } from "../index";
+
+dayjs.extend(customParseFormat);
+
+export const checkReferrer = (
+    referrerHeader: string | undefined,
+    requestFromPath: string,
+    errorPath: string,
+    res: NextApiResponse,
+) => {
+    if (!referrerHeader?.includes(requestFromPath)) {
+        redirectTo(res, errorPath);
+    }
+};
+
+export const validateSummary = (summary: string, errors: ErrorInfo[], errorId: string) => {
+    const summaryRequiredErr: ErrorInfo = {
+        id: errorId,
+        errorMessage: "Please enter a Summary",
+    };
+
+    if (!requireFieldCheck(summary)) {
+        errors.push(summaryRequiredErr);
+    } else {
+        const lengthErr: ErrorInfo = {
+            id: errorId,
+            errorMessage: "Please enter a Summary of less than 100 characters",
+        };
+
+        if (!validateLength(summary, 100)) {
+            errors.push(lengthErr);
+        }
+    }
+};
+
+export const validateDisruptionType = (
+    disruptionType: "planned" | "unplanned" | undefined,
+    errors: ErrorInfo[],
+    errorId: string,
+) => {
+    const distruptionTypeRequiredErr: ErrorInfo = {
+        id: errorId,
+        errorMessage: "Please choose a Type of Disruption",
+    };
+
+    if (!requireFieldCheck(disruptionType)) {
+        errors.push(distruptionTypeRequiredErr);
+    } else {
+        const invalidTypeErr: ErrorInfo = {
+            id: errorId,
+            errorMessage: "Invalid Disruption Type Selected. Please choose a valid Type of Disruption",
+        };
+
+        if (!validateDisruptionTypeValue(disruptionType, DISRUPTION_TYPES)) {
+            errors.push(invalidTypeErr);
+        }
+    }
+};
+
+export const requireFieldCheck = (field: string | undefined): boolean => {
+    let checkPassed = false;
+    if (field) {
+        //errors.push(error);
+        checkPassed = true;
+    }
+
+    return checkPassed;
+};
+
+export const validateDisruptionTypeValue = (
+    value: "planned" | "unplanned" | undefined,
+    validArray: string[],
+): boolean => {
+    let isValid = false;
+    if (value && validArray.includes(value)) {
```

Comment:
Added the array to avoid updating the function when there is a value change. For example when a new disruption type is added inline with open close principle

Reply:
But we could just have a constant in this function which holds the types of disruptions which are valid.

If this function is the function responsible for holding the types of disruption which are valid, then it could just be held in this function.

Reply:
Removed this method and added a method named `isValueInArray` in `apiUtils/index.js`. This method checks if a value is present in an values array and can be reused.

Reply:
As per another comment, we need to remove the isValueInArray function as it is built into javascript.